### PR TITLE
Implement login feature

### PR DESCRIPTION
**Pull request by hothanhtuan1209**
#19 
In this pull request i build login functionality, use custom model, use email and password for authentication, send alternate email for username.
image
image
image

### CODE REVIEW

File: myproject/supermarket_manager/views.py [hunk 1]
```diff
@@ -384,3 +388,39 @@ def get_account_detail(request, account_id):
...
+    response. If the login credentials are incorrect, it returns a 401 Unauthorized response.
+
+    Parameters:
+        request (HttpRequest): The HTTP request object containing user login credentials.
+
+    Returns:
+        JsonResponse: A JSON response indicating the result of the login attempt.
+    """
+
+    if request.method == "POST":
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py]: Limit nesting of conditional statements.

File: myproject/supermarket_manager/views.py [hunk 2]
```diff
@@ -384,3 +388,39 @@ def get_account_detail(request, account_id):
...
+
+
+@csrf_protect
+def login_account(request):
+    """
+    API endpoint for user login.
+
+    This function handles POST requests for user login. Users can log in using their email
+    and password. If either the email or password is missing, it returns a 400 Bad Request
+    response. If the login credentials are incorrect, it returns a 401 Unauthorized response.
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py]: login not `log in`.
Should be separated into bullet points.

File: README.md [hunk 3]
```diff
@@ -18,22 +18,36 @@ This is a project to build mini supermarket management software, including Accou
...
-5. Running server
- >py manage.py server
+5. Create role
+> py manage.py create_role
+
+6. Create permission
+> py manage.py create_permissions
+
+7. Create superuser
+> py manage.py createsuperuser
```
Comment (cuongdoan-agilityio) [inline README.md]: Can we just use 1 custom command?

File: myproject/supermarket_manager/management/commands/create_permissions.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from django.core.management.base import BaseCommand
+from supermarket_manager.models import Permission
+
+class Command(BaseCommand):
+    help = "Create custom permission"
+
+    def handle(self, *args, **kwargs):
+        permission_need_create = [
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/management/commands/create_permissions.py]: I think you should use another constant name.

File: myproject/supermarket_manager/models.py [hunk 5]
```diff
@@ -51,14 +52,41 @@ def __str__(self):
...
+    This manager provides methods for creating user accounts and superuser accounts.
+
+    Attributes:
+        BaseUserManager: The base manager class provided by Django.
+    """
+    
+    def create_user(self, email, user_name, password=None, **extra_fields):
+        if not email:
+            raise ValueError('The Email field must be set')
+        email = self.normalize_email(email)
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/models.py:68]: Should add empty line.

File: myproject/supermarket_manager/views.py (hunk 1 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

File: myproject/supermarket_manager/views.py (hunk 2 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

File: README.md (hunk 3 above)
Comment (hothanhtuan1209) [reply README.md]: Done

File: myproject/supermarket_manager/management/commands/create_permissions.py (hunk 4 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/management/commands/create_permissions.py]: Done

File: myproject/supermarket_manager/models.py (hunk 5 above)
Comment (hothanhtuan1209) [inline myproject/supermarket_manager/models.py:69]: Done

File: myproject/supermarket_manager/views.py [hunk 6]
```diff
@@ -384,3 +388,39 @@ def get_account_detail(request, account_id):
...
+
+    if not email or not password:
+        return JsonResponse({"message": REQUIRED_LOGIN}, status=400)
+
+    user = authenticate(request, email=email, password=password)
+
+    if user is not None:
+        login(request, user)
+        return JsonResponse({"message": LOGIN}, status=200)
+    else:
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py]: No need `else` statement here.

Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

**Pull request merged**